// This is the Protobuf definition for RepoCommandParameter, described at:
// http://redmine.named-data.net/projects/repo-ng/wiki/Repo_Command
// Compile this file using:
// protoc --cpp_out=. repo-command-parameter.proto
syntax = "proto3";
package ndn_message;

// Protobuf has no "outer" message type, so we need to put the TLV RepoCommandParameter
// message inside an outer "typeless" RepoCommandParameterMessage.
message RepoCommandParameterMessage {
  message Name {
    repeated bytes component = 8;
  }

  message Selectors {
    uint32 min_suffix_components = 13;
    uint32 max_suffix_components = 14;
    bytes publisher_public_key_locator = 15;
    // TODO: Add Exclude.
    uint32 child_selector = 17;
  }

  message RepoCommandParameter {
    Name name = 7;
    Selectors selectors = 9;
    uint64 start_block_id = 204;
    uint64 end_block_id = 205;
    uint64 process_id = 206;
    uint64 max_interest_num = 211;
    uint64 watch_timeout = 212;
    uint64 watch_status = 213;
    uint64 interest_lifetime = 214;
  }

  // A RepoCommandParameterMessage has one RepoCommandParameter.
  RepoCommandParameter repo_command_parameter = 201;
}
